sdk/attestation/azure-security-attestation/src/main/java/com/azure/security/attestation/AttestationClientBuilder.java

```diff
@@ -233,53 +331,75 @@ public AttestationClientBuilder tokenValidationOptions(AttestationTokenValidatio
         return this;
     }
 
-    /**
-     * Builds an instance of AttestationClient sync client.
-     *
-     * Instantiating a synchronous Attestation client:
-     * <br>
-     * <!-- src_embed com.azure.security.attestation.AttestationClientBuilder.buildClient -->
-     * <pre>
-     * AttestationClient client = new AttestationClientBuilder&#40;&#41;
-     *     .endpoint&#40;endpoint&#41;
-     *     .buildClient&#40;&#41;;
-     * </pre>
-     * <!-- end com.azure.security.attestation.AttestationClientBuilder.buildClient -->
-     * @return an instance of {@link AttestationClient}.
-     */
-    public AttestationClient buildClient() {
-        return new AttestationClient(buildAsyncClient());
-    }
-
-    /**
-     * Builds an instance of AttestationAsyncClient async client.
-     *
-     * Instantiating a synchronous Attestation client:
-     * <br>
-     * <!-- src_embed com.azure.security.attestation.AttestationClientBuilder.buildAsyncClient -->
-     * <pre>
-     * AttestationAsyncClient asyncClient = new AttestationClientBuilder&#40;&#41;
-     *     .endpoint&#40;endpoint&#41;
-     *     .buildAsyncClient&#40;&#41;;
-     * </pre>
-     * <!-- end com.azure.security.attestation.AttestationClientBuilder.buildAsyncClient -->
-     * @return an instance of {@link AttestationClient}.
-     */
-    public AttestationAsyncClient buildAsyncClient() {
-        return new AttestationAsyncClient(buildInnerClient(), this.tokenValidationOptions);
-    }
-
     /**
      * Builds an instance of AttestationClientImpl with the provided parameters.
      *
      * @return an instance of AttestationClientImpl.
      */
     private AttestationClientImpl buildInnerClient() {
-        AttestationServiceVersion version = serviceVersion != null ? serviceVersion : AttestationServiceVersion.getLatest();
-        clientImplBuilder.apiVersion(version.getVersion());
-        if (tokenCredential != null) {
-            clientImplBuilder.addPolicy(new BearerTokenAuthenticationPolicy(tokenCredential, dataplaneScope));
+
+//        AttestationClientImplBuilder clientImplBuilder = new AttestationClientImplBuilder();
+        // Global Env configuration store
+        Configuration buildConfiguration = (configuration == null)
+            ? Configuration.getGlobalConfiguration()
+            : configuration;
+
+        // Service version
+        AttestationServiceVersion version = serviceVersion != null
+            ? serviceVersion
+            : AttestationServiceVersion.getLatest();
+
+        // endpoint cannot be null, which is required in request authentication
+        String endpoint = this.endpoint;
+        Objects.requireNonNull(endpoint, "'Endpoint' is required and can not be null.");
+
+        // If the customer provided a pipeline, use it, otherwise configure the pipeline based on the options
+        // which were provided.
+        HttpPipeline pipeline = this.pipeline;
```

Comment:
Other than the user-agent bug what else motivated you to create your own httppipeline here?

Reply:
I will make sure we capture those to make our inner builders better.

Reply:
It was entirely the user-agent bug. I was also following the examples used by all the other Java SDKs that I looked at.